pkg/pipelines/tekton_checks_test.go

```diff
@@ -0,0 +1,135 @@
+package pipelines
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/fake"
+)
+
+func TestTektonChecker(t *testing.T) {
+	var tests = []struct {
+		desscription       string
+		existingCRDs       []runtime.Object
+		wantedResult       bool
+		wantedErrorMessage string
+	}{
+		{"All required CRDs are present.",
+			[]runtime.Object{
+				crd("pipelines.tekton.dev"),
+				crd("pipelineresources.tekton.dev"),
+				crd("pipelineruns.tekton.dev"),
+				crd("triggerbindings.tekton.dev"),
+				crd("triggertemplates.tekton.dev"),
+				crd("clustertasks.tekton.dev"),
+				crd("taskruns.tekton.dev"),
+				crd("tasks.tekton.dev"),
+			},
+			true,
+			"",
+		},
+		{"All required CRDs are present with some extra CRDs",
+			[]runtime.Object{
+				crd("pipelines.tekton.dev"),
+				crd("pipelineresources.tekton.dev"),
+				crd("pipelineruns.tekton.dev"),
+				crd("triggerbindings.tekton.dev"),
+				crd("triggertemplates.tekton.dev"),
+				crd("clustertasks.tekton.dev"),
+				crd("taskruns.tekton.dev"),
+				crd("tasks.tekton.dev"),
+				crd("something else.tekton.dev"),
+			},
+			true,
+			"",
+		},
+		{"Missed one required pipeline CRD",
+			[]runtime.Object{
+				crd("pipelineresources.tekton.dev"),
+				crd("pipelineruns.tekton.dev"),
+				crd("triggerbindings.tekton.dev"),
+				crd("triggertemplates.tekton.dev"),
+				crd("clustertasks.tekton.dev"),
+				crd("taskruns.tekton.dev"),
+				crd("tasks.tekton.dev"),
+			},
+			false,
+			"",
+		},
+		{"Missed one required trigger CRD",
+			[]runtime.Object{
+				crd("pipelines.tekton.dev"),
+				crd("pipelineresources.tekton.dev"),
+				crd("pipelineruns.tekton.dev"),
+				crd("triggertemplates.tekton.dev"),
+				crd("clustertasks.tekton.dev"),
+				crd("taskruns.tekton.dev"),
+				crd("tasks.tekton.dev"),
+			},
+			false,
+			"",
+		},
+		{"Missed more than one required CRDs",
+			[]runtime.Object{
+				crd("pipelineresources.tekton.dev"),
+				crd("pipelineruns.tekton.dev"),
+				crd("triggertemplates.tekton.dev"),
+				crd("clustertasks.tekton.dev"),
+				crd("taskruns.tekton.dev"),
+				crd("tasks.tekton.dev"),
+			},
+			false,
+			"",
+		},
+		{"Zero required CRDs",
+			[]runtime.Object{},
+			false,
+			"",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desscription, func(t *testing.T) {
+
+			tektonChecker, err := newFakeChecker(test.existingCRDs)
+			if err != nil {
+				t.Fatal(err)
+			}
+			result, err := tektonChecker.checkInstall()
+			if result != test.wantedResult {
+				t.Fatalf("Want check result '%v', but got '%v'", test.wantedResult, result)
```

Comment:
These are table tests, better to use `Errorf` which won't quit after the first failure.

This applies below too.